Python/Product/Debugger/Debugger/TaskExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using static System.FormattableString;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.PythonTools.Debugger {
+    class TaskExtensions {
+        private static readonly Lazy<IVsThreadedWaitDialogFactory> _twdf = new Lazy<IVsThreadedWaitDialogFactory>(() => {
+            var twdf = (IVsThreadedWaitDialogFactory)Package.GetGlobalService(typeof(SVsThreadedWaitDialogFactory));
+            if (twdf == null) {
+                var err = Invariant($"{nameof(SVsThreadedWaitDialogFactory)} is not available");
+                Trace.Fail(err);
+                throw new InvalidOperationException(err);
```

Comment:
>InvalidOperationException [](start = 26, length = 25)

This will likely crash the entire process if it occurs. Better to just not show the dialog.